secboot/secboot_sb_test.go

```diff
@@ -967,6 +967,15 @@ func (s *secbootSuite) TestResealKey(c *C) {
 			TPMPolicyAuthKeyFile: mockTPMPolicyAuthKeyFile,
 		}
 
+		numMockSealedKeyObjects := len(myParams.KeyFiles)
```

Comment:
Actually, I'm not sure why this change is needed and what exactly it's doing - if I change to the original code (or git revert d52049308d3122b89ed25804459123a94abae703):
```
diff --git a/secboot/secboot_sb_test.go b/secboot/secboot_sb_test.go
index 4992c039de..6cbc50bb89 100644
--- a/secboot/secboot_sb_test.go
+++ b/secboot/secboot_sb_test.go
@@ -967,15 +967,6 @@ func (s *secbootSuite) TestResealKey(c *C) {
                        TPMPolicyAuthKeyFile: mockTPMPolicyAuthKeyFile,
                }
 
-               numMockSealedKeyObjects := len(myParams.KeyFiles)
-               mockSealedKeyObjects := make([]*sb_tpm2.SealedKeyObject, 0, numMockSealedKeyObjects)
-               for range myParams.KeyFiles {
-                       mockSealedKeyFile := filepath.Join("test-data", "keyfile")
-                       mockSealedKeyObject, err := sb_tpm2.ReadSealedKeyObject(mockSealedKeyFile)
-                       c.Assert(err, IsNil)
-                       mockSealedKeyObjects = append(mockSealedKeyObjects, mockSealedKeyObject)
-               }
-
                sequences := []*sb_efi.ImageLoadEvent{
                        {
                                Source: sb_efi.Firmware,
@@ -1042,6 +1033,7 @@ func (s *secbootSuite) TestResealKey(c *C) {
 
                // mock ReadSealedKeyObject
                readSealedKeyObjectCalls := 0
+               mockSealedKeyObjects := []*sb_tpm2.SealedKeyObject{{}, {}}
                restore = secboot.MockSbReadSealedKeyObject(func(keyfile string) (*sb_tpm2.SealedKeyObject, error) {
                        readSealedKeyObjectCalls++
                        c.Assert(keyfile, Equals, myParams.KeyFiles[readSealedKeyObjectCalls-1])
```
afaict this test is still fine. But maybe/probably I'm missing something?

Reply:
@mvo so yes, i had a version that did `mockSealedKeyObjects := []*sb_tpm2.SealedKeyObject{{}, {}}` which @anonymouse64 rejected.

Please make up your mind if you want {} empty mocks, or realistic mocks, which are not needed here at all.

Reply:
See https://github.com/snapcore/snapd/pull/10715#pullrequestreview-748340551

Reply:
Sorry, I did miss the backstory here, I think it's fine then, what Ian says makes sense (but it's also very heavily mocked so I would be ok with something simpler) anyway, let's not bike-shed this here, I will make my comment "resolved".